pkg/apis/experimental/v1alpha1/custom_builder.go

```diff
@@ -0,0 +1,39 @@
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	duckv1alpha1 "knative.dev/pkg/apis/duck/v1alpha1"
+
+	"github.com/pivotal/kpack/pkg/apis/build/v1alpha1"
+)
+
+func (b *CustomBuilder) BuildBuilderSpec() v1alpha1.BuildBuilderSpec {
+	return v1alpha1.BuildBuilderSpec{
+		Image:            b.Status.LatestImage,
+	}
+}
+
+// FIXME : is this correct?
+func (b *CustomBuilder) Image() string {
+	return b.Spec.Tag
```

Comment:
Hmm. This is only used in the metadata retriever. I wonder if we can make it so that the `BuilderResource` does not have this method. Instead the metadata retriever can use an interface that embeds and extends the BuilderResource?

Reply:
We can do this, curious why a custom builder doesn't need to have an Image conceptually though. Even if we don't use that method, it seems to fulfill that interface.

Reply:
It certainly could implement the interface. (Probably return `b.Status.LatestImage`?)

I think the value in not including the `Image` method would be in being explicit about our intentions and not writing methods we don't need. A little interface-segregation principle.

Reply:
I buy that. I'm going to think a bit about the BuilderResource interface in general.